packages/ua-devtools-solana/src/oft/sdk.ts

```diff
@@ -124,17 +127,110 @@ export class OFT extends OmniSDK implements IOApp {
 
     @AsyncRetriable()
     async getEnforcedOptions(eid: EndpointId, msgType: number): Promise<Bytes> {
+        // First we check that we can understand the message type
+        this.assertMsgType(msgType)
+
         const eidLabel = `eid ${eid} (${formatEid(eid)})`
+        this.logger.verbose(`Getting enforced options for ${eidLabel} and message type ${msgType}`)
+
+        try {
+            const options = await OftTools.getEnforcedOptions(this.connection, this.configAccount, eid, this.publicKey)
+            const optionsForMsgType = msgType === MSG_TYPE_SEND ? options.send : options.sendAndCall
```

Comment:
I know you know this, but just wanted to point out that if we ever add further message types in the default implementation, we will need to update this.

Reply:
I know, this just works with what I have from the monorepo and that function only allows these two, hence these checks